### DISCUSSION TITLE
Question: Random seeder set everytime at __getitem__ ?

### DISCUSSION THREAD

**Discussion by SynthAether** [coqui-ai/TTS#2088] [General Q&A] [answered]
Hi, 
Sorry for not including a working code, but doing a code inspection I noticed the random seeder is set everytime at `__getitem__`

https://github.com/coqui-ai/TTS/blob/f237e4ccd9f2fd2b0cb5e136dfe0e20cc32bf898/TTS/vocoder/datasets/gan_dataset.py#L76

This makes me suspect there is less variety when randomly selecting sub-segments from the training data

https://github.com/coqui-ai/TTS/blob/f237e4ccd9f2fd2b0cb5e136dfe0e20cc32bf898/TTS/vocoder/datasets/gan_dataset.py#L144

cheers
Shaun

**Comment by erogol** [comment, marked as answer]
Thanks for the flag. It sets every time but not to the same value.

**Reply by SynthAether** [reply]
Thanks, the assumption here that torch.utils.data.get_worker_info().seed will be different on every call.
